web-client/extras/lib/BufferUtils.ts

```diff
@@ -0,0 +1,363 @@
+import { SerialBuffer } from './SerialBuffer';
+import { StringUtils } from './StringUtils';
+
+type TypedArray =
+    | Int8Array
+    | Uint8Array
+    | Int16Array
+    | Uint16Array
+    | Int32Array
+    | Uint32Array
+    | Uint8ClampedArray
+    | Float32Array
+    | Float64Array;
+
+class BufferUtils {
+    static BASE64_ALPHABET = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/';
+    static BASE32_ALPHABET = {
+        RFC4648: 'ABCDEFGHIJKLMNOPQRSTUVWXYZ234567=',
+        RFC4648_HEX: '0123456789ABCDEFGHIJKLMNOPQRSTUV=',
+        NIMIQ: '0123456789ABCDEFGHJKLMNPQRSTUVXY',
+    };
+    static HEX_ALPHABET = '0123456789abcdef';
+
+    static _BASE64_LOOKUP: string[] = [];
+    private static _ISO_8859_15_DECODER?: TextDecoder | null;
+    private static _UTF8_ENCODER?: TextEncoder | null;
+
+    static toAscii(buffer: Uint8Array): string {
+        const chunkSize = 0x2000;
+        const buf = BufferUtils._toUint8View(buffer);
+
+        let ascii = '';
+        for (let i = 0; i < buf.length; i += chunkSize) {
+            ascii += String.fromCharCode.apply(null, [...buf.subarray(i, i + chunkSize)]);
+        }
+        return ascii;
+    }
```

Comment:
This isn't `toAscii`, it's more like `toIso8559_1`. It doesn't throw on bytes >= 128.